alchemiscale/storage/objectstore.py

```diff
@@ -264,22 +274,23 @@ def pull_protocoldagresult(
             The ProtocolDAGResult corresponding to the given `ProtocolDAGResultRef`.
 
         """
-        if transformation.scope != protocoldagresult.scope:
-            raise ValueError(
-                f"transformation scope '{transformation.scope}' differs from protocoldagresult scope '{protocoldagresult.scope}'"
-            )
-
         route = "results" if ok else "failures"
 
-        # build `location` based on gufe key
-        location = os.path.join(
-            "protocoldagresult",
-            *protocoldagresult.scope.to_tuple(),
-            transformation.gufe_key,
-            route,
-            protocoldagresult.gufe_key,
-            "obj.json",
-        )
+        # build `location` based on provided ScopedKey if not provided
+        if location is None:
+            if transformation.scope != protocoldagresult.scope:
```

Comment:
Should also check that `transformation` and `protocoldagresult` are not both None as is default. Ie can get this situation 
```python
location == None 
transformation == None 
protocoldagresult == None
```
Which will have a fit when trying to check 

```python
if transformation.scope != protocoldagresult.scope:
```

Reply:
Ah good call; adding in a guardrail here.